src/freertos_drivers/ti/CC32xxUart.cxx

```diff
@@ -113,28 +119,55 @@ void CC32xxUart::disable()
     MAP_UARTDisable(base);
 }
 
-/** Try and transmit a message.
+/** Send data until there is no more space left.
  */
-void CC32xxUart::tx_char()
+void CC32xxUart::send()
 {
-    if (txPending == false)
+    do
     {
         uint8_t data = 0;
-
         if (txBuf->get(&data, 1))
         {
-            if (txEnableAssert)
-            {
-                txEnableAssert();
-            }
             MAP_UARTCharPutNonBlocking(base, data);
 
-            MAP_IntDisable(interrupt);
-            txPending = true;
-            MAP_UARTIntEnable(base, UART_INT_TX);
-            MAP_IntEnable(interrupt);
-            txBuf->signal_condition();
         }
+        else
+        {
+            break;
+        }
+    }
+    while (MAP_UARTSpaceAvail(base));
+
+    if (txBuf->pending())
+    {
+        /* more data to send later */
+        MAP_UARTTxIntModeSet(base, UART_TXINT_MODE_FIFO);
+    }
+    else
+    {
+        /* no more data left to send */
+        MAP_UARTTxIntModeSet(base, UART_TXINT_MODE_EOT);
+    }
+}
+
+/** Try and transmit a message.
+ */
+void CC32xxUart::tx_char()
+{
+    if (txPending == false)
```

Comment:
This is not required because the interrupt mode will be set appropriately the next time the CC32xxUart::send() method is called from the interrupt handler.